build_projects/shared-build-targets-utils/VersionRepoUpdater.cs

```diff
@@ -81,31 +94,33 @@ private static List<NuGetPackageInfo> GetPackageInfo(string nupkgFilePath)
 
         private async Task UpdateGitHubFile(string path, string newFileContent, string commitMessage)
         {
+            await PushChangesToGitHubFile(path, Encoding.UTF8.GetBytes(newFileContent), commitMessage);
+        }
+
+        private async Task PushChangesToGitHubFile(string githubFile, byte[] data, string commitMessage)
+        {
             using (HttpClient client = new HttpClient())
             {
                 client.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json");
                 client.DefaultRequestHeaders.Add("Authorization", $"token {_gitHubAuthToken}");
                 client.DefaultRequestHeaders.Add("User-Agent", _gitHubUser);
 
-                string fileUrl = $"https://api.github.com/repos/{_versionsRepoOwner}/{_versionsRepo}/contents/{path}";
+                string fileUrl = $"https://api.github.com/repos/{_versionsRepoOwner}/{_versionsRepo}/contents/{githubFile}";
 
-                Console.WriteLine($"Getting the 'sha' of the current contents of file '{_versionsRepoOwner}/{_versionsRepo}/{path}'");
+                Console.WriteLine($"Getting the 'sha' of the current contents of file '{_versionsRepoOwner}/{_versionsRepo}/{githubFile}'");
 
                 string currentFile = await client.GetStringAsync(fileUrl);
                 string currentSha = JObject.Parse(currentFile)["sha"].ToString();
 
                 Console.WriteLine($"Got 'sha' value of '{currentSha}'");
 
-                Console.WriteLine($"Request to update file '{_versionsRepoOwner}/{_versionsRepo}/{path}' contents to:");
```

Comment:
This is actually really helpful when trying to diagnose issues when the script fails.  Can we attempt to keep this logging somehow?  (Maybe blindly UTF8 convert the data, and even if is garbage print it out?)